packages/analytics/client/src/analytics_client/analytics_client.ts

```diff
@@ -103,8 +113,27 @@ export class AnalyticsClient implements IAnalyticsClient {
       );
     }
 
+    // If the validator is registered (dev-mode only), perform the validation.
+    if (eventTypeOpts.validator) {
+      validateSchema(`Event Type '${eventType}'`, eventTypeOpts.validator, eventData);
+    }
+
+    const event: Event = {
+      timestamp,
+      event_type: eventType,
+      context: this.context$.value,
+      properties: eventData,
+    };
+
+    // debug-logging before checking the opt-in status to help during development
     if (this.initContext.isDev) {
-      // TODO: In the future we may need to validate the eventData based on the eventType's registered schema (only if isDev)
+      this.initContext.logger.debug(`Report event "${eventType}"`);
+      // If in the browser, log the event as such (to make it easier to debug in the browser's console).
+      //   It needs the casting because the logger types expect debug to be a string
+      // If in the server, stringify to avoid printing [object Object]
+      this.initContext.logger.debug(
+        global.window ? (event as unknown as string) : JSON.stringify(event)
+      );
```

Comment:
Unrelated to the original issue: I took advantage of this PR to add a debug logger that will print the events. It will help developers to check how their events look like when they implement them :)